compress/decompress.go

```diff
@@ -682,20 +692,33 @@ func (g *Getter) Match(buf []byte) (bool, uint64) {
 			g.dataP++
 			g.dataBit = 0
 		}
-		if lenBuf != 0 {
+		if lenBuf != 0 || lenBuf != int(wordLen) {
 			g.dataP, g.dataBit = savePos, 0
 		}
-		return lenBuf == int(wordLen), g.dataP
+		if lenBuf == int(wordLen) {
+			return 0
+		}
+		if lenBuf < int(wordLen) {
+			return -1
+		}
+		if lenBuf > int(wordLen) {
+			return 1
```

Comment:
seems forgot to restore internal state of cursor here
something like `g.dataP, g.dataBit = savePos, 0`

in my head - pattern of Match usage looks like: 
```
for getter.HasNext() {
   if getter.Match(prefix) {
       key, _ := getter.Next(nil)
       value, _ := getter.Next(nil)
       // process key/value ....
 }
}
```

Reply:
So the Match shouldn't move cursor to the next position if buf matches the word in file?

Reply:
I don't know. Maybe you right (naming is always hard).
Have 2 different methods is fine for me. 

Then I need MatchPrefix to return `-1,0,1`. But there is 2 corner-cases: match of prefix (`ab` on `abcd`) and full-match of key (`abcd` on `abcd`). Maybe `MatchPrefix` can return `(cmp int, fullMatchOfKey bool, nextOffset uint64)`? We doing binary-search by this method in Erigon3.